### PR TITLE
Rendering search bar

### PR DESCRIPTION
**Pull request by KarinLiegmann**
Hi, I would be glad if you could take a look at my code.
I added a SearchBar on the RecipeResults-page (which is currently rendered as the landing-page of the app). In there, you can search for ingredients and get autocomplete-suggestion after typing in 3 characters (currently set to 2 autocomplete-suggestions). The autocomplete-suggestions are fetched in real time from an external recipe-api.
The results are currently saved in LOCAL STORAGE (under the keys "ingredients" and "activeIngredients").

You can add the ingredient either by pressing Enter while in the input-field or by clicking on the green plus-sign on the right side of the searchbar. 
You will only be able to add ingredients if there is a corresponding match on the API. Otherwise you will get an error message.

You can choose the autocomplete-suggestion by clicking on it and then pressing Enter/Add Button. If you choose to type out the full name and ignore the autocomplete-suggestion (e.g. you choose to type "chicken" even though the autosuggestion "chicken" is rendered) you will only be able to add the ingredient if they are a complete match. this ensures that it is still a valid ingredient.

Thank you in advance!

### CODE REVIEW

**Review by mhstrkmp — COMMENTED** [review]
good work 🚀

**Comment by KarinLiegmann** [comment]
Wow, that was fast! Thank you very much!!

File: client/src/components/SearchBar.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+    const getQueryValue = (event) => {
+        const value = event.target.value
+        const query = value.toLowerCase()
+        setSearchQuery(query)
+    }
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        if (searchQuery.length >= 3 && ingredient.length !== 0 && isError !== true) {
```
Comment (miriamhaenle) [inline client/src/components/SearchBar.js:23]: I think you can shorten the last query to `!isError`as `isErrror`is a boolean
```suggestion
        if (searchQuery.length >= 3 && ingredient.length !== 0 && !isError) {
```

**Review by miriamhaenle — CHANGES_REQUESTED** [review]
Hi Karin, 
this looks really good. I only added a few comments / suggestions :)

File: client/src/App.js [hunk 2]
```diff
@@ -1,18 +1,63 @@
...
 
   const [open, setOpen] = useState(false)
 
+  const [ingredients, setIngredients] = useState(loadFromLocal('ingredients') ?? [])
+  const [activeIngredients, setActiveIngredients] = useState(loadFromLocal('activeIngredients') ?? [])
+
+  function addIngredient(ingredient) {
+    const newIngredient =
+    {
+      ingredientName: ingredient.name,
```
[the remark this answers is not in the document]
Comment (KarinLiegmann) [reply client/src/App.js]: Thank you, you are right. I changed the name :)

File: client/src/components/SearchBar.js (hunk 1 above)
Comment (KarinLiegmann) [reply client/src/components/SearchBar.js]: Thank you for the great tipp!

File: client/src/App.js [hunk 3]
```diff
@@ -1,18 +1,63 @@
...
+    saveToLocal('ingredients', ingredients)
+    saveToLocal('activeIngredients', allActiveIngredients)
+  }
+
+  useEffect(() => {
+    filterActiveIngredients()
+  }, [ingredients])
+
   return (
     <div className="App">
```
[the remark this answers is not in the document]
Comment (KarinLiegmann) [reply client/src/App.js:38]: Actually I have used the App-className in my GlobalStyles.js to give the App some basic styling. I think it looks less cluttered than using Styled-Components in the App.js directly. For now, I will leave the className, but thank you very much for the suggestion!

**Pull request merged**